Onboarding refactor

HactarCE opened the pull request:
<img width="1648" height="976" alt="Screenshot 2025-10-07 at 6 57 20 PM" src="https://github.com/user-attachments/assets/ae7289c0-8820-4fdf-ae28-84fb6bd64942" />

Fixes #39347

Release Notes:

- Improved onboarding UI by collapsing it to a single page

dinocosta committed:
chore(gpui): change default window size

Change default window size from 1024x700 pixels to 1536x864 pixels.

dinocosta committed:
chore(onboarding): remove navbar

As part of the new onboarding redesign, this commit removes the navbar
that was presented on the left side of the onboarding page, that allowed
the user to select different pages for the onboarding settings. Although
the pages code and functionality is still present in the codebase, these
will no longer be rendered and only the basics page will be rendered.

dinocosta committed:
refactor(onboarding): move start building button to top of the page

This commit takes the "Start Building" button that used to live in the
onboarding's left side navbar and moves it to the default onboarding
screen, to the top right corner as outlined in the designs.

dinocosta committed:
chore(onboarding): move import settings section to the default page

This commit moves the "Import Settings" section to the the top left
corner of the onboarding screen. It also does some minor redesigning of
that section, in regards to the label size.

Worth noting that this does not yet follow the finalized design but gets
us closer to it.

dinocosta committed:
chore(onboarding): remove keybindings to navigate between pages

Update the default keymaps for Linux, MacOS and Windows to remove the
keybindings that would navigate between the onboarding's pages. Since
the actions are no longer used, those have been removed too.

dinocosta committed:
chore(onboarding): remove ai_setup_page module and related code

- Remove `onboarding::ai_setup_page` module - Remove `AiSetup` as a
valid value for the `onboarding::SelectedPage` enum. - Update code
according to the removals above.

dinocosta committed:
refactor(onboarding): move import settings section to basics_page

This commit moves the "Import Settings" section to the
`onboarding::basics_page` module so as to allow us to completely remove
the `onboarding::editing_page` module.

Doing this breaks the design a little bit, as the "Finish Setup" button
is no longer in the same div as the "Import Settings" section, but it's
something we can fix later. The issue is that for the "Finish Setup"
button we need a reference to the `window` and `scroll_handle`, which we
don't currently have and might no need to propagate, if we eventually
move all these `render_x_section` functions to the `onboarding` module
itself.

dinocosta committed:
refactor(onboarding): move import settings section to basics_page

This commit moves the "Import Settings" section to the
`onboarding::basics_page` module, as well as the "Finish Setup" button
too.

Co-authored-by: Lukas Wirth <me@lukaswirth.dev>

dinocosta committed:
chore(onboarding): remove editing_page module

dinocosta committed:
refactor(onboarding): add zed logo and update import settings look

HactarCE committed:
Make button styling more flexible

Co-authored-by: Mikayla Maki <mikayla.c.maki@gmail.com>
Co-authored-by: Anthony Eid <hello@anthonyeid.me>

HactarCE committed:
Polish onboarding page

Co-authored-by: Mikayla Maki <mikayla.c.maki@gmail.com>
Co-authored-by: Anthony Eid <hello@anthonyeid.me>

HactarCE committed:
fix clippy

Co-authored-by: Mikayla Maki <mikayla.c.maki@gmail.com>

HactarCE committed:
use outlined transparent button style in onboarding

Co-authored-by: Mikayla Maki <mikayla.c.maki@gmail.com>

dinocosta committed:
fix(ui): ensure that border is rendered for outlined transparent

Update the `ButtonLike.render` implementation to ensure that a border of
1 pixel is rendered for the `OutlinedTransparent` style.

dinocosta committed:
Merge branch 'main' into onboarding-refactor

dinocosta committed:
fix(onboarding): remove unused dependencies

This commit will please cargo-machete!

dinocosta commented:
The fix for the "Failed to migrate settings" error is here – https://github.com/zed-industries/zed/pull/39771 . We're going to merge this and, after it's merged, I'll merge `main` back into this PR ✅

dinocosta committed:
Merge branch 'main' into onboarding-refactor

dinocosta committed:
chore(onboarding): remove selected page enum and state

mikayla-maki committed:
Tidy up the PR for merging

Co-authored-by: HactarCE <6060305+HactarCE@users.noreply.github.com>
Co-authored-by: dino <dinojoaocosta@gmail.com>

mikayla-maki committed:
Merge branch 'main' into onboarding-refactor

mikayla-maki committed:
WIP: Make onboarding telemetry and diagnostics section match the designs

dinocosta committed:
fix(onboarding): add missing type annotations and please clippy

- Add missing type annotations for when `SwitchField::new` has no title
- Fix clippy's complaints

mikayla-maki committed:
Fix a bug where keybindings wouldn't be drawn immediately
Normalize to non-period sentences

mikayla-maki committed:
fmt

mikayla-maki committed:
Fix doc tests
